libhttpsimple-bindings/src/main/scala/libhttpsimple/LibHttpSimple.scala

```diff
@@ -158,12 +158,11 @@ object LibHttpSimple {
       result
     }
 
-  private def httpHeadersToDelimitedString(headers: Map[String, String]): String =
+  private def httpHeadersToDelimitedString(headers: Map[String, String]): Seq[String] =
     headers
       .map {
         case (headerName, headerValue) => s"$headerName$HttpHeaderNameAndValueSeparator $headerValue"
-      }
-      .mkString(CRLF)
+      }.toList
```

Comment:
Just an aside -- Vectors perform better in the typical case. List works well when you don't need to know the length and are building it up via prepending elements, usually in simple tail recursive functions. Either works here, just thought I'd mention.